### ISSUE TITLE
Should we require tests and CI for every package?

### ISSUE THREAD

**Issue by mitar** [Meteor-Community-Packages/organization#18]
It think that if we expect not main maintainers to help maintain the package (like to release critical bugfixes in time, or update to new Meteor releases), we should maybe require all packages to have tests and CI. So that it is easy to check if at least the main base case works.

**Comment by copleykj** [comment]
I'm on board with this.

**Comment by StorytellerCZ** [comment]
Would be great. Would be nice to create a basic CI example with the tests and publication. We'll need it for every package anyway.

**Comment by mitar** [comment]
I have [this package](https://github.com/peerlibrary/meteor-classy-test) which makes it easy to do such things, and also to have tests which have a sequence interleaving server and client side.

**Comment by mitar** [comment]
I also requested GitHub CI so that we can also use that for packages instead of external services.

**Comment by copleykj** [comment]
What is the best method of testing packages at this time? Trying to find the most relevant and up to date information is currently quite troublesome.

**Comment by mitar** [comment]
For packages, I am still using `meteor test-packages`. It works pretty well for me (especially using the package I linked above). I do not know if anything better was made in meantime.

**Comment by copleykj** [comment]
Yeah,  `meteortesting` has a mocha package but I seem to remember something about issues using it with the `testpackages` command.

I wanna dig into your classy test package. Just gotta mentally prepare myself to wade through the coffeescript. 😂

**Comment by SimonSimCity** [comment]
I would again have to dig into it, but I have some working examples of testing packages using [meteor-mocha](https://github.com/meteortesting/meteor-mocha) and doing code-coverage using [meteor-coverage](https://github.com/serut/meteor-coverage).

All this is available in my latest version of meteor-roles, which already has been transferred to this repo 😊I've just slightly adjusted what @serut proposed in his repo - so the credit goes to him for this draft: https://github.com/serut/meteor-coverage#specific-setup-for-meteor-package

Here's my config which (in addition to the one proposed by @serut) directly produces an HTML output of the code-coverage:

```json
{
  "scripts": {
    "setup-test": "rm -rf ./someapp && meteor create --minimal someapp && cd someapp && cp ../.coverage.json . && meteor npm i --save puppeteer && mkdir packages && ln -s ../../ ./packages/roles",
    "test": "npm run setup-test && cd someapp && TEST_BROWSER_DRIVER=puppeteer COVERAGE=1 COVERAGE_OUT_JSON=1 COVERAGE_APP_FOLDER=$(pwd)/ meteor test-packages --once --raw-logs --driver-package meteortesting:mocha ./packages/roles; istanbul report --include .coverage/summary.json --root . --dir .coverage",
    "test:watch": "cd someapp && TEST_WATCH=1 COVERAGE=1 COVERAGE_APP_FOLDER=$(pwd)/ meteor test-packages --raw-logs --driver-package meteortesting:mocha ./packages/roles",
    "todobeforepublish": "rm -rf .npm/ node_modules/ someapp/"
  },
  "dependencies": {
    "istanbul": "^0.4.5"
  }
}
```

**Comment by SimonSimCity** [comment]
In my opinion we should highly recommend these two packages to every developer. The only reason I see going away from meteor-mocha is if you want all test-results in a browser (see: https://github.com/meteortesting/meteor-mocha/issues/70).

As we're on it, we should maybe also work put some work down on the package ... specially looking at https://github.com/meteortesting/meteor-mocha/issues/80 which also requests an update of the meteor-guide for how to test.

**Comment by StorytellerCZ** [comment]
I think it would be best to add testing and CI to [template package](https://github.com/Meteor-Community-Packages/template-package) and then have the existing packages slowly progress in that direction.
